sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java

```diff
@@ -214,6 +215,26 @@ public void finish(FinishBundleContext c) {
       beamSchema = Schema.builder().addField("payload", Schema.FieldType.BYTES).build();
       valueMapper = getRawBytesToRowFunction(beamSchema);
 
+    } else if (format != null && format.equals("PROTO")) {
+      String fileDescriptorPath = configuration.getFileDescriptorPath();
+      String messageName = configuration.getMessageName();
+
+      if (messageName == null) {
+        throw new IllegalArgumentException(
+            "To read from PubSubLite in PROTO format, messageName must be provided.");
+      }
```

Comment:
nit: might make sense to have a `validate()` method in the Configuration class that can have checks like this. (and could move the other checks like the one for `inputSchema` below)